cpp/include/ray/api/function_manager.h

```diff
@@ -153,27 +230,48 @@ class FunctionManager {
   }
 
   template <typename Function>
-  bool RegisterRemoteFunction(std::string const &name, const Function &f) {
-    /// Now it is just support free function, it will be
-    /// improved to support member function later.
-    auto pair = func_ptr_to_key_map_.emplace((uint64_t)f, name);
+  absl::enable_if_t<!std::is_member_function_pointer<Function>::value, bool>
+  RegisterRemoteFunction(std::string const &name, const Function &f) {
+    auto pair = func_ptr_to_key_map_.emplace(GetAddress(f), name);
     if (!pair.second) {
       return false;
     }
 
     return RegisterNonMemberFunc(name, f);
   }
 
+  template <typename Function>
+  absl::enable_if_t<std::is_member_function_pointer<Function>::value, bool>
+  RegisterRemoteFunction(std::string const &name, const Function &f) {
+    auto pair = func_ptr_to_key_map_.emplace(GetAddress(f), name);
+    if (!pair.second) {
+      return false;
```

Comment:
Can we throw a exception when `RAY_REMOTE` failed? Maybe user don't want to do a judge about this return value.